### DISCUSSION TITLE
Issue with allowReconnection Not Working on Second Attempt

### DISCUSSION THREAD

**Discussion by ibeallc** [colyseus/colyseus#734] [General]
Hello everyone,

I'm encountering an issue with the allowReconnection function in my game server code. The function works correctly for the initial reconnection attempt, but fails on subsequent attempts. Here’s the relevant part of my onLeave method:
` 
async onLeave(client: Client, consented: boolean) {
  let self = this;

  // Find the player based on session ID
  let player = this.findPlayerBySessionId(client.sessionId);
  if (player) {
    // Mark the player as away
    player.away = true;
  } else {
    console.log('Player not found');
    return;
  }

  try {
    if (consented) {
      // Handle consented leave
      self.removePlayersInfo(client.sessionId);
      this.rmPlayerFromGame(client.sessionId);
      self.activePlayers--;
      throw new Error('consented leave');
    }

    // Try to allow reconnection within 5 minutes
    await this.allowReconnection(client, 60 * 5);

    // If reconnection is successful, mark player as not away
    let plyr = this.findPlayerBySessionId(client.sessionId);
    if (plyr) {
      plyr.away = false;
      self.updatePlayerReconnectionToken(client);
    } else {
      console.log('Player not found after reconnection');
    }
  } catch (e) {
    // Handle unsuccessful reconnection
    console.log('Reconnection not successful', e);
    self.activePlayers--;

    // Update lobby metadata
    self.setMetadata({
      activePlayers: self.activePlayers
    }).then(() => updateLobby(self));

    // Remove the player from the game
    this.rmPlayerFromGame(client.sessionId);
  }
}
`
**Issues Observed:**

- The first reconnection attempt works fine.
- On the second reconnection attempt, allowReconnection does not seem to function as expected and the player is not able to reconnect.

**Error Message:**

❌ reconnection token invalid or expired. Did you miss .allowReconnection()?
👉 https://docs.colyseus.io/server/room/#allowreconnection-client-seconds

**Comment by endel** [comment]
Hi @ibeallc, each connection and reconnection generates a new `.reconnectionToken` in the client-side. Are you always using the latest one available? Cheers

**Reply by ibeallc** [reply]
I thought so too, so I tried saving the token again with `self.updatePlayerReconnectionToken(client);` when the player reconnects the first time. However, the `client. reconnectionToken` still seems to be the same.

**Reply by ibeallc** [reply]
Oh, sorry, I should have retrieved the token again on the client side. I tried this in the onLeave method.

**Reply by ibeallc** [reply]
Thanks to your ultra pro instinct reply! :))))
